public/app/features/dashboard-scene/settings/DashboardEditView.tsx

```diff
@@ -0,0 +1,60 @@
+import { useLocation } from 'react-router-dom';
+
+import { locationUtil, NavModelItem } from '@grafana/data';
+import { SceneObject } from '@grafana/scenes';
+import { t } from 'app/core/internationalization';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { useSelector } from 'app/types';
+
+import { DashboardScene } from '../scene/DashboardScene';
+
+import { AnnotationsEditView } from './AnnotationsEditView';
+import { GeneralSettings } from './GeneralSettings';
+import { VariablesEditView } from './VariablesEditView';
+
+export interface DashboardEditView extends SceneObject {
+  getUrlKey(): string;
+}
+
+export function useDashboardEditPageNav(dashboard: DashboardScene, currentEditView: string) {
+  const location = useLocation();
+  const navIndex = useSelector((state) => state.navIndex);
+  const navModel = getNavModel(navIndex, 'dashboards/browse');
+  const dashboardPageNav = dashboard.getPageNav(location, navIndex);
+
+  const pageNav: NavModelItem = {
+    text: 'Settings',
+    children: [
+      {
+        text: t('dashboard-settings.general.title', 'General'),
+        url: locationUtil.getUrlForPartial(location, { editview: 'settings', editIndex: null }),
+        active: currentEditView === 'settings',
+      },
+      {
+        text: t('dashboard-settings.annotations.title', 'Annotations'),
+        url: locationUtil.getUrlForPartial(location, { editview: 'annotations', editIndex: null }),
+        active: currentEditView === 'annotations',
+      },
+      {
+        text: t('dashboard-settings.variables.title', 'Variables'),
+        url: locationUtil.getUrlForPartial(location, { editview: 'variables', editIndex: null }),
+        active: currentEditView === 'variables',
+      },
```

Comment:
I guess we're missing here the rest of the pages
```suggestion
      },
      // TODO: Add Links, Versions and JSON Model
```

Reply:
yes, this was just an initial test of the structure